packages/peregrine/lib/talons/CartPage/PriceAdjustments/ShippingMethods/useShippingFields.js

```diff
@@ -0,0 +1,115 @@
+ * step in the process. We need to be very mindful that these values are never
+ * displayed to the user.
+ */
+const MOCKED_ADDRESS = {
+    city: 'city',
+    firstname: 'firstname',
+    lastname: 'lastname',
+    street: ['street'],
+    telephone: 'telephone'
+};
+
+export const useShippingFields = props => {
+    const {
+        getCountriesQuery,
+        getStatesQuery,
+        setIsFetchingMethods,
+        setShippingMutation
+    } = props;
+    const formApi = useFormApi();
+    const { values: formValues } = useFormState();
+    const { country, state, zip } = formValues;
+    const [{ cartId }] = useCartContext();
+
+    const {
+        data: countriesData,
+        error: countriesError,
+        loading: isCountriesLoading
+    } = useQuery(getCountriesQuery);
+
+    const [
+        fetchStates,
+        {
+            called: fetchStatesCalled,
+            data: statesData,
+            error: statesError,
+            loading: isStatesLoading
+        }
+    ] = useLazyQuery(getStatesQuery);
+
+    const [setShippingAddress, { loading: setShippingLoading }] = useMutation(
+        setShippingMutation
+    );
+
+    useEffect(() => {
+        if (country) {
+            fetchStates({
+                variables: {
+                    countryCode: country
+                }
+            });
+        }
+    }, [country, fetchStates, formApi]);
+
+    useEffect(() => {
+        setIsFetchingMethods(setShippingLoading);
+    }, [setIsFetchingMethods, setShippingLoading]);
+
+    const handleZipOnBlur = useCallback(() => {
+        if (country && state && zip) {
+            setShippingAddress({
+                variables: {
+                    cartId,
+                    address: {
+                        ...MOCKED_ADDRESS,
+                        country_code: country,
+                        postcode: zip,
+                        region: state
+                    }
+                }
+            });
+        }
+    }, [cartId, country, setShippingAddress, state, zip]);
+
+    let formattedCountriesData = [{ label: 'Loading Countries...', value: '' }];
+    if (!isCountriesLoading && !countriesError) {
+        const { countries } = countriesData;
+        formattedCountriesData = countries.map(country => ({
+            label: country.full_name_english,
+            value: country.two_letter_abbreviation
+        }));
+        formattedCountriesData.sort((a, b) => (a.label < b.label ? -1 : 1));
+    }
+
+    let formattedStatesData = [];
+    if (fetchStatesCalled && !isStatesLoading && !statesError) {
+        const { country } = statesData;
+        const { available_regions: availableRegions } = country;
+        if (availableRegions) {
+            formattedStatesData = availableRegions.map(region => ({
+                id: region.id,
+                label: region.name,
+                value: region.code
+            }));
+            formattedStatesData.unshift({ label: '', value: '' });
```

Comment:
You do this to present an empty state initially, right? Rather than allowing it as an option maybe we should set the initial selection to `<option disabled selected value> -- Select a FOO -- </option>` inside the `Select` component as a default (or configurable) UI.

Reply:
Oh, good catch. I intended to add support for `disabled` and `hidden` props to the Select component options and just completely forgot, will get those added.

Reply:
Added these props in 9c94499